src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -139,6 +139,15 @@ public async Task<MetadataAsSourceFile> GetGeneratedFileAsync(Project project, I
                         }
                         catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
                         {
+                            // Insert the exception message at the top of the fallback Metadata As Source document to
+                            // help with diagnosing failures. When the generated source is added to the document, this
+                            // comment will end up at the bottom.
+                            var failureText = e.ToString();
+                            var failureTextAsComment = "// " + failureText.Replace("\n", "\n// ") + "\n";
+
+                            var sourceText = await temporaryDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                            temporaryDocument = temporaryDocument.WithText(sourceText.Replace(new TextSpan(0, 0), failureTextAsComment));
```

Comment:
The file is empty up to this point, right? So the `new TextSpan(0,0)` doesn't really matter what it is before? Add a comment?

Reply:
It might or might not be empty at this point. The comment already says it's placed at the top (which can be seen) and ends up at the bottom (covered by tests).

Reply:
When is it not already empty?

Reply:
"Insert this at the top" implies there's something after our insertion point, but the test doesn't show anything after. Either we can add an assert here that the existing file is empty, or we're missing a test case.

Reply:
I'm not aware of a case where it's not empty here. However, the code added by this pull request adheres to the stated behavior whether or not the document is empty at this point.